### DISCUSSION TITLE
Mouse behaviour question / layouts

### DISCUSSION THREAD

**Discussion by lfrancke** [leftwm/leftwm#444] [Q&A] [answered]
Hi,

I'm evaluating LeftWM and I like what I see so far, thank you for the work on it.
I'm coming from bspwm which might shape my expectations so I just might be doing things "wrong".

I can resize windows with the right mouse button but every time I do so it _also_ turns windows into floating windows. In bspwm the same functionality does exist but it stays in tiling mode and just changes the size of the window being dragged in relation to the others.
I couldn't find a really good documentation on how layouts are supposed to work in LeftWM. Are they "static" or are they supposed to be changeable?
Example: MainAndVertStack layout splits both columns evenly but by resizing a window I'd like to change the width of the main window to take (e.g.) 3/4 of the size.

The second issue I have is when dragging windows using the mouse I have a hard time getting it to "click" into place back into tiling mode. By moving the mouse just millimeters I can sometimes see things "flicker" and then I can release and it works but it's definitely not smooth.

I'm wondering if this is me or "intended".

I didn't want to open issues for these as I'm not sure about the intended behaviour.

Thank you!

**Comment by VuiMuich** [comment, marked as answer]
Welcome Adventurer and thanks for trying out LeftWM.

Currently we have only fully automated dynamic window sizes in tiling mode. As you have noticed resizing a window automatically puts it in floating mode.
Some layouts (I believe only the LeftWiderAndStack (will Edit in the proper ones later)) have sizing theming options, and we have a command to adjust margins on the fly. Besides that we had no reqests like yours so far, iirc. But If you like you are welcome to open a feature request for this, as I can imagine this might be actually helpful, but probably is not a quick fix.

I agree with the second issue you mention, that dragging windows back to tiling is a bit wacky. A bit of a workaround might be the keybindable command `FloatingToTile` (again not sure if correct name, will edit later). But again, you are welcome to open an issue on that one, as this truly deserves some improvement.

Thank you for your feedback!

**Reply by lfrancke** [reply]
Thank you very much for the response.

That helps a lot. I wasn't even sure what the intended behavior is supposed to be so I wasn't sure if it's a feature request or a bug.
I understand much better now.

I'll try to remember to open issues for both of these.
